icu4c/source/i18n/tibetancal.h

```diff
@@ -0,0 +1,313 @@
+// © 2016 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+/*
+ *****************************************************************************
+ * Copyright (C) 2003-2021, International Business Machines Corporation
+ * and others. All Rights Reserved.
+ *****************************************************************************
+ *
+ * File TIBETANCAL.H
+ *****************************************************************************
+ */
+
+#ifndef TIBETANCAL_H
+#define TIBETANCAL_H
+
+#include "unicode/utypes.h"
+
+#if !UCONFIG_NO_FORMATTING
+
+#include "unicode/calendar.h"
+
+U_NAMESPACE_BEGIN
+
+/**
+*  Concrete class which provides the Tibetan calendar.
+*
+*
+*
+*
+*
+*
+*
+*/
+
+
+class U_I18N_API TibetanCalendar : public Calendar {
+public:
+  /**
+   * Useful constants for TibetanCalendar.
+   * @internal
+   */
+
+
+  /**
+   * Calendar type - phugpa or tsurphu
+   * @internal 
+   */
+  enum ECalculationType {
+    PHUGPA,
+    TSURPHU
+  };
+
+  //-------------------------------------------------------------------------
+  // Constructors...
+  //-------------------------------------------------------------------------
+
+  /**
+   * Constructs an TibetanCalendar based on the current time in the default time zone
+   * with the given locale.
+   *
+   * @param aLocale  The given locale.
+   * @param success  Indicates the status of TibetanCalendar object construction.
+   *                 Returns U_ZERO_ERROR if constructed successfully.
+   * @param type     The Tibetan calendar calculation type. The default vaue is PHUGPA.
+   * @internal
+   */
+  TibetanCalendar(const Locale& aLocale, UErrorCode &success, ECalculationType type = PHUGPA);
+
+
+  /**
+   * Copy Constructor
+   * @internal
+   */
+  TibetanCalendar(const TibetanCalendar& other);
+
+
+  /**
+   * Destructor.
+   * @internal
+   */
+  virtual ~TibetanCalendar();
+
+
+  /**
+   * Sets Tibetan calendar calculation type used by this instance.
+   *
+   * @param type    The calendar calculation type, <code>PHUGPA</code> to use the phugpa
+   *                calendar, <code>TSURPHU</code> to use the tsurphu calendar.
+   * @internal
+   */
+  void setCalculationType(ECalculationType type, UErrorCode &status);
+
+
+  /**
+   * Returns <code>true</code> if this object is using the phugpa
+   * calendar, or <code>false</code> if using the tsurphu calendar.
+   * @internal
+   */
+  UBool isPhugpa();
+
+
+  // clone
+  virtual TibetanCalendar* clone() const;
+
+
+
+  /**
+   * return the calendar type, "tibetan-phugpa" or "tibetan-tsurphu".
+   *
+   * @return calendar type
+   * @internal
+   */
+  virtual const char * getType() const;
+
+
+private:
+
+  //-------------------------------------------------------------------------
+  // Internal data....
+  //-------------------------------------------------------------------------
+
+/**
+ * Return the moonTab by using the table for the given integer value. 
```

Comment:
nit: Return->Returns (please also change the rest of the comments).
Please explain what is moonTab.

Reply:
why do moonTab and sunTab need to be a methood on this class. It seems the implementation is not depending on any member value of the TibetanCaledar, right? Could we remove them from the private and just implement it as C style helper function internally inside tibetancal.cpp instead?